doc/sqa/sdd/ravenSoftwareDesing.tex

```diff
@@ -0,0 +1,483 @@
+model will be interpreted/used by the framework and, at run-time, will become part of RAVEN itself.
+
+\paragraph{Reduced Order Model} ~\\
+ A ROM (also called Surrogate Model) is a mathematical representation of a system, used to predict a selected output 
+ space of a physical system.
+The ``training'' is a process that uses sampling of the physical model to improve the prediction capability (capability to 
+predict the status of the system given a realization of the input space) of the ROM. More specifically, in RAVEN the 
+Reduced Order Model is trained to emulate a high fidelity numerical representation (system codes) of the physical system. 
+Two general characteristics of these models can be generally assumed (even if exceptions are possible):
+\begin{enumerate}
+   \item The higher the number of realizations in the training sets, the higher is the accuracy of the prediction performed by 
+   the reduced order model. This statement is true for most of the cases although some ROMs might be subject to the overfitting issues. The over-fitting phenomenon is not discussed here, since its occurrence is highly dependent on the 
+   algorithm type, (and there is large number of ROM options available in RAVEN). Every time the user chooses a particular 
+   reduced order model algorithm to use, he should consult the relevant literature;
+   \item The smaller the size of the input domain with respect to the variability of the system response, the more likely the 
+   surrogate model will be able to represent the system output space.
+\end{enumerate}
+
+
+\paragraph{Hybrid Models} ~\\
+The Hybrid Model is able to combine ROM and any other high-fidelity Model (e.g. Code, ExternalModel). 
+The ROMs will be 
+``trained'' based on the results from the high-fidelity model. The accuracy of the ROMs will be evaluated based on the 
+cross validation scores, and the validity of the ROMs will be determined via some local validation metrics. After these 
+ROMs are trained, the HybridModel can decide which of the Model (i.e the ROMs or  high-fidelity model) to be executed 
+based on the accuracy and validity of the ROMs.
+
+\paragraph{Ensemble Models} ~\\
+The Ensemble Model is aimed to create a chain of Models (whose execution order is determined by the Input/Output 
+relationships among them). If the relationships among the models evolve in a non-linear system, a Picard’s Iteration 
+scheme is employed.
+
+\paragraph{PostProcessor} ~\\
+The Post-Processor model represents the container of all the data analysis capabilities in the RAVEN code. This model is 
+aimed to process the data (for example, derived from the sampling of a physical code) in order to identify representative F
+Figure of Merits. For example, RAVEN owns Post-Processors for performing statistical and regression/correlation analysis, 
+data mining and clustering, reliability evaluation, topological decomposition, etc.
+
+\subsubsection{Distribution}
```

Comment:
The question is raised before, you may add ```s``` for consistence. If you agree, please update the whole documents for other entities.